notebook/notebook/notebook.py

```diff
@@ -355,27 +375,15 @@ def notebook_status_from_pod(pod):
     ready = (notebook['state'] == 'Ready')
     count = 0
     while count < 10:
-        status = await notebook_status_from_notebook(k8s, service, headers, cookies, notebook)
-        if not status:
-            async with dbpool.acquire() as conn:
-                async with conn.cursor() as cursor:
-                    await cursor.execute(
-                        'DELETE FROM notebooks WHERE user_id = %s;',
-                        userdata['id'])
-            ready = False
-            break
-        if status['state'] != notebook['state']:
-            async with dbpool.acquire() as conn:
-                async with conn.cursor() as cursor:
-                    await cursor.execute(
-                        'UPDATE notebooks SET state = %s, pod_ip = %s WHERE user_id = %s;',
-                        (status['state'], status['pod_ip'], userdata['id']))
-            ready = (status['state'] == 'Ready')
+        new_status = await notebook_status_from_notebook(k8s, service, headers, cookies, notebook)
+        changed = await update_notebook_return_changed(dbpool, user_id, notebook, new_status)
```

Comment:
Want to check that if an exception happens in `k8s_notebook_status_from_notebook` (there is a raise if status isn't 404) we still delete the notebook in `update_notebook_return_changed`, rather than exiting which could leave a dangling record. I've a bit about this, and know there are cases where the exception isn't returned (and we would get a default None for new_status), and that one case for this is in using `ensure_future`, which we're not using.

Reply:
I don't think we should delete the notebook unless we've verified it is gone by getting a 404.  What if k8s gets temporarily overloaded and we get a timeout exception?  We should try again later, not delete the notebook.

I do agree we should catch (and discard) errors in the loop.  Otherwise, we return to the client who refreshes, potentially getting into a fast refresh loop.  I pushed a fix.